### ISSUE TITLE
New config api (decouple tile type from icon)

### ISSUE THREAD

**Issue by moinmoin-sh** [moinmoin-sh/OXRS-MM-WallPanel-ESP32-FW#33]
The goal of this redesign is to decouple the tile type (behavior) from the icon. In principle each tile type can have each icon.

This list shows available tile types , possible icons, use cases, basic behavior

type | icon | possible use | behavior
-- | -- | -- | --
button | any | light | button tiles
button_level_up | any | brightness/volume
button_level_down | any | blinds
button_text | (text) | custom
button_number | (number) | custom
indicator_text | (text) | message/content | indicator only
indicator_number | (number) | Temp./rH/time
indicator_icon | any | state display: door/window
mediaplayer | play/pause | player | special tile
dropdown | none | select from list | calling popups
keypad | any | access control
remote | any | remote
link | any | change screen | jump to screen

The input group per tile in he AdminUi wouldlook like this:
(no flags anymore)

![grafik](https://user-images.githubusercontent.com/53935853/174683388-330af56f-2a3e-4aec-9da5-cdaea7dda820.png)

**Comment by sumnerboy12** [comment]
This looks really good Moin.

**Comment by plains203** [comment]
I am wondering if the Lock Screen poppy can have a flag that stops you from closing it? It could be set a different way than the admin page though.

**Comment by sumnerboy12** [comment]
The popup is only closed if/when the backend sends the `close` command. So it can be configured to be left visible always (once launched).

**Comment by sumnerboy12** [comment]
Ah yes, my apologies, you are right - that is still the case so you could still close it currently.

**Comment by plains203** [comment]
> Ah yes, my apologies, you are right - that is still the case so you could still close it currently.

All good, I just double checked in case a change had been made that I missed.

**Comment by plains203** [comment]
The comment has been made on the discord re being able to potentially upload our own icon arrays to specify against the tiles. Is that something that might be possible and worth thinking about in this rework?

**Comment by plains203** [comment]
> > I am wondering if the Lock Screen poppy can have a flag that stops you from closing it? It could be set a different way than the admin page though.
> 
> I would prefer having 2 different tile types for the keypad such as `keypad `and `keypad-blocking` which could make sense for the lock screen as mentioned in an other issue as well.

Yeah that would probably be a better way of handling it. I assume you would do the blocking screen as something you could call using cmnd instead of having to click on it so that you could programmatically lock the device.

**Comment by moinmoin-sh** [comment]
> The comment has been made on the discord re being able to potentially upload our own icon arrays to specify against the tiles. Is that something that might be possible and worth thinking about in this rework?

The decoupling of types from icons is the first step into that direction. I think about a standard set of icons builtin the FW and somehow offer a path for custom icons - propably via mqtt.

**Comment by plains203** [comment]
Yeah that sounds like a solid plan to me.
